README.md

```diff
@@ -8,92 +8,196 @@
 ```
 
-## `@transform`
+## `@orderby`
 
-Add additional columns based on keyword arguments.
+Sort rows in a `DataFrame` by values in one of several columns or a 
+transformation of columns. Only operates on `DataFrame`s and not `GroupedDataFrame`s. 
 
 ```julia
-@transform(df, newCol = cos.(:x), anotherCol = :x.^2 + 3*:x .+ 4)
+df = DataFrame(x = [1, 1, 2, 2], y = [1, 2, 101, 102]);
+@orderby(df, -1 .* :x)
+@orderby(df, :x, :y .- mean(:y))
+```
+
+## `@with`
+
+`@with` creates a scope in which all symbols that appear are aliases for the columns
+in a DataFrame. 
+
+```julia
+df = DataFrame(x = 1:3, y = [2, 1, 2])
+x = [2, 1, 0]
+
+@with(df, :y .+ 1)
+@with(df, :x + x)  # the two x's are different
+
+x = @with df begin
+    res = 0.0
+    for i in 1:length(:x)
+        res += :x[i] * :y[i]
+    end
+    res
+end
+
+@with(df, df[:x .> 1, ^(:y)]) # The ^ means leave the :y alone
+
 ```
 
 !!! note 
-    
-    Versions of DataFrames.jl `0.19` and above support the operators `Between`, `All`,
-    and `Not` when selecting and transforming columns. DataFramesMeta.jl does not currently
-    support this syntax. 
+    `@with` creates a function, so scope within `@with` is a local scope.
+    Variables in the parent can be read. Writing to variables in the parent scope
+    differs depending on the type of scope of the parent. If the parent scope is a
+    global scope, then a variable cannot be assigned without using the `global` keyword.
+    If the parent scope is a local scope (inside a function or let block for example),
+    the `global` keyword is not needed to assign to that parent scope.
+
+!!! note
+    Because `@with` creates a function, be careful with the use of `return`. 
+
+    ```
+    function data_transform(df; returnearly = false)
+        if returnearly
+            @with df begin 
+                z = :x + :y
+                return z
+            end
+        else 
+            return [1, 2, 3]
+        end
+
+        return [4, 5, 6]
+    end
+    ```
 
+    The above function will return `[4, 5, 6]` because the `return` inside the `@with`
+    applies to the anonymous function created by `@with`. 
 
-## `@byrow`
+    Given that `@eachrow` (below) is implemented with `@with`, the same caveat applies to 
+    `@eachrow` blocks. 
 
-Act on a DataFrame row-by-row. Includes support for control flow and `begin end` blocks. Since the "environment" induced by `@byrow df` is implicitly a single row of `df`, one uses regular operators and comparisons instead of their elementwise counterparts as in `@with`. Does not change the input data frame argument.
+
+## `@eachrow`
+
+Act each row of a data frame. Includes support for control flow and `begin end` blocks. Since the "environment" induced by `@eachrow df` is implicitly a single row of `df`, one uses regular operators and comparisons instead of their elementwise counterparts as in `@with`. Does not change the input data frame argument.
```

Comment:
```suggestion
Act on each row of a data frame. Includes support for control flow and `begin end` blocks. Since the "environment" induced by `@eachrow df` is implicitly a single row of `df`, one uses regular operators and comparisons instead of their elementwise counterparts as in `@with`. Does not change the input data frame argument.
```